client/driver/java.go

```diff
@@ -137,33 +143,42 @@ func (d *JavaDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle,
 		args = append(args, driverConfig.Args...)
 	}
 
-	// Setup the command
-	// Assumes Java is in the $PATH, but could probably be detected
-	execCtx := executor.NewExecutorContext(d.taskEnv)
-	cmd := executor.Command(execCtx, "java", args...)
-
-	// Populate environment variables
-	cmd.Command().Env = d.taskEnv.EnvList()
-
-	if err := cmd.Limit(task.Resources); err != nil {
-		return nil, fmt.Errorf("failed to constrain resources: %s", err)
+	bin, err := discover.NomadExecutable()
+	if err != nil {
+		return nil, fmt.Errorf("unable to find the nomad binary: %v", err)
 	}
 
-	if err := cmd.ConfigureTaskDir(d.taskName, ctx.AllocDir); err != nil {
-		return nil, fmt.Errorf("failed to configure task directory: %v", err)
+	pluginLogFile := filepath.Join(ctx.AllocDir.AllocDir, "plugin.out")
+	pluginConfig := &plugin.ClientConfig{
+		Cmd: exec.Command(bin, "executor", pluginLogFile),
 	}
 
-	if err := cmd.Start(); err != nil {
-		return nil, fmt.Errorf("failed to start source: %v", err)
+	exec, pluginClient, err := createExecutor(pluginConfig, d.config.LogOutput)
+	if err != nil {
+		return nil, err
+	}
+	executorCtx := &executor.ExecutorContext{
```

Comment:
Actually we can set those for all platforms now. On platforms where isolation and resource constraints isn't possible it would be a no-op.